sqlx-core/Cargo.toml

```diff
@@ -151,7 +151,7 @@ sha-1 = { version = "0.9.8", default-features = false, optional = true }
 sha2 = { version = "0.9.8", default-features = false, optional = true }
 sqlformat = "0.1.8"
 thiserror = "1.0.30"
-time = { version = "0.2.27", optional = true }
+time = { version = "0.3.2", features = ["macros", "formatting", "parsing"], optional = true }
```

Comment:
Wouldn't `#[serde(with = "time::serde::rfc3339")]` (as described in [this comment](https://github.com/launchbadge/sqlx/pull/1455#issuecomment-1042296608)) require the [`serde-well-known` feature flag](https://docs.rs/time/0.3.9/time/serde/index.html)? Since serializing with RFC3339 format is a very common use case, I guess adding `serde-well-known` would make sense...

Reply:
Generally you don't want to enable feature flags you don't strictly need. That would force that feature on for all crates that depend on SQLx.

Reply:
Makes sense... but what is then the recommended way to read a timestamp from the database and put it into a DTO in RFC3339 format? Since the version of `time` that sqlx comes with doesn't support this (due to version and that feature flag), a user of sqlx has to use an intermediate "DatabaseDTO" with `sqlx::types::time::OffsetDateTime` for fetching from the database (using `query_as`), just to then convert it manually into an "ActualDTO" with `time::OffsetDateTime` (I did that with `time::OffsetDateTime::from_unix_timestamp(db_dto.created_at.unix_timestamp())`, but I'm not sure that's the best way). Only then the aforementioned way to serialize a timestamp using RFC3339 with `with` actually works. Or am I missing something?

Reply:
That's only an issue because SQLx is currently using a different version of `time` than you are. Otherwise `sqlx::types::time::OffsetDateTime` and `time::OffsetDateTime` should always refer to the same type.

Reply:
> `time::serde::rfc3339` exists since `time` v0.3.6, so it's not available in the version proposed in this PR

sqlx depending on `0.3.2` means versions `>=0.3.2, <0.4.0` are allowed, so you're free to use features from newer patch versions of the time crate

Reply:
The version specified here is not restrictive and Cargo will automatically pull in the latest `0.3.x` release if it's not in your `Cargo.lock` already when building your crate. 

You should also specify `time = { version =  "0.3.6", features = ["serde-well-known"] }` in your own dependencies to be certain of this.